File: app/src/main/java/me/ykrank/s1next/view/adapter/AppPostListRecyclerViewAdapter.java
package me.ykrank.s1next.view.adapter;

import android.app.Activity;
import android.support.annotation.NonNull;

import com.github.ykrank.androidtools.ui.adapter.delegate.FooterProgressAdapterDelegate;

import me.ykrank.s1next.data.api.app.model.AppThread;
import me.ykrank.s1next.view.adapter.delegate.AppPostAdapterDelegate;

/**
 * This {@link android.support.v7.widget.RecyclerView.Adapter}
 * has another item type {@link FooterProgressAdapterDelegate}
 * in order to implement pull up to refresh.
 */
public final class AppPostListRecyclerViewAdapter extends BaseRecyclerViewAdapter {
    @NonNull
    private AppPostAdapterDelegate postAdapterDelegate;

    public AppPostListRecyclerViewAdapter(Activity activity) {
        super(activity);

        postAdapterDelegate = new AppPostAdapterDelegate(activity);
        addAdapterDelegate(postAdapterDelegate);
        addAdapterDelegate(new FooterProgressAdapterDelegate(activity));
    }

    public void setThreadInfo(@NonNull AppThread threadInfo) {
        postAdapterDelegate.setThreadInfo(threadInfo);
    }
}

File: app/src/main/java/me/ykrank/s1next/view/adapter/AppPostListRecyclerViewAdapter.kt
package me.ykrank.s1next.view.adapter

import android.app.Activity

import com.github.ykrank.androidtools.ui.adapter.delegate.FooterProgressAdapterDelegate

import me.ykrank.s1next.data.api.app.model.AppThread
import me.ykrank.s1next.view.adapter.delegate.AppPostAdapterDelegate

/**
 * This [android.support.v7.widget.RecyclerView.Adapter]
 * has another item type [FooterProgressAdapterDelegate]
 * in order to implement pull up to refresh.
 */
class AppPostListRecyclerViewAdapter(activity: Activity, quotePid: String?) : BaseRecyclerViewAdapter(activity) {

    private val postAdapterDelegate: AppPostAdapterDelegate = AppPostAdapterDelegate(activity, quotePid)

    init {
        addAdapterDelegate(postAdapterDelegate)
    }

    fun setThreadInfo(threadInfo: AppThread) {
        postAdapterDelegate.setThreadInfo(threadInfo)
    }
}
